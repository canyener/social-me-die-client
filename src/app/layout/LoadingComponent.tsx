import React from 'react'
import { Dimmer, Loader } from 'semantic-ui-react'

const LoadingComponent: React.FC<{ inverted?: boolean, content?: string }> = ({ inverted, content }) => {
  return (
    <Dimmer active inverted={inverted}>
      <Loader content={content} />
    </Dimmer>
  )
}

export default LoadingComponent