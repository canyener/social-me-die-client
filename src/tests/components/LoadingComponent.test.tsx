import React from 'react'
import { shallow } from 'enzyme'

import LoadingComponent from '../../app/layout/LoadingComponent'

test('Should render LoadingComponent correctly with given inverted and content props', () => {
  const wrapper = shallow(
    <LoadingComponent 
      inverted={true}
      content="Loading stuff"
    />
  )

  expect(wrapper).toMatchSnapshot()
})

test('Should render LoadingComponent correctly without any given props', () => {
  const wrapper = shallow(
    <LoadingComponent />
  )

  expect(wrapper).toMatchSnapshot()
})