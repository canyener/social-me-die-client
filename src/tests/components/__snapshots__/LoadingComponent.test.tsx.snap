// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoadingComponent correctly with given inverted and content props 1`] = `
<Dimmer
  active={true}
  inverted={false}
>
  <Loader
    content="Loading stuff"
  />
</Dimmer>
`;

exports[`Should render LoadingComponent with inverted as true and no content without any given props 1`] = `
<Dimmer
  active={true}
  inverted={true}
>
  <Loader />
</Dimmer>
`;
